node/actors/network/src/gossip/attestation_status.rs

```diff
@@ -0,0 +1,57 @@
+use std::fmt;
+
+use zksync_concurrency::sync;
+use zksync_consensus_roles::attester;
+
+use crate::watch::Watch;
+
+/// Coordinate the attestation by showing the status as seen by the main node.
+#[derive(Debug, Clone)]
+pub struct AttestationStatus {
+    /// Next batch number where voting is expected.
+    ///
+    /// Its value is `None` until the background process polling the main node
+    /// can establish a value to start from.
+    pub next_batch_to_attest: Option<attester::BatchNumber>,
```

Comment:
are we 100% sure `None` should be part of the state space? Is it possible to delay initialization before it's accessible?

Reply:
In https://github.com/matter-labs/zksync-era/pull/2544 even the API can return `None` until `Genesis` is available. I'm not sure how long that is, probably not that long in practice. I'm also finding it difficult to understand what a delayed initialisation would mean for an external node: it doesn't run until the main node API is available _and_ returns `Some`, okay, but how about the main node itself? If I can't start the `Executor` until we have `Some` value for this, can it even create the `Genesis` in the first place?

Reply:
If the genesis.first_block points to a block in the future, then it may happen that the initial batch number is not known. Since now the made the AttestationStatusWatch the API, we can require that it is initialized before the executor is started (i.e.wait for the first full response from the attestationStatus rpc).

Reply:
In https://github.com/matter-labs/zksync-era/pull/2544 API may return None, because I don't understand the design principles behind the Web3Error, so I wasn't sure how to embed the NotFound response in the error space.

Reply:
Okay, if it is indeed safe to delay starting the Executor on the main node until we have this information by polling the store, we can get rid of the None from the Watch. 

On the external node it’s safe because it will have to pull genesis anyway

I will do it as a separate PR.